### ISSUE TITLE
LoadRatingPlan performance

### ISSUE THREAD

**Issue by ewsamuels** [cgrates/cgrates#41]
HI DanB,
As per our discussions on the IRC channel when calling LoadRatingPlan with 170K unique destinations it takes 45mins to 1 hour for that process to complete so this ticket has been opened at your request.
I believe you were also testing with my data and using shared destination and it was still running even after 35 mins.

Regards,
Errol

**Comment by danbogos** [comment]
Hi Errol,

We have implemented a new API which should serve you faster in your specific case. Please check ApierV1.LoadTariffPlanFromStorDb. It should match the speed of loading from .csv files now.

Please reopen if performance issue is still there.

Thanks,
DanB

**Comment by ewsamuels** [comment]
Hi DanB,
Do you have a link to the documentation so I know what parameters we need to pass for testing?
I wasn't able to locate it here: http://godoc.org/github.com/cgrates/cgrates/apier/v1

Thanks
Errol

**Comment by ewsamuels** [comment]
Ok nevermind I found the info I needed in the commit.

**Comment by ewsamuels** [comment]
Hi DanB,
I was finally able to test this from clean install using the package dated 9th April but the results are not very consistent.
I tested this a few times and on average it took approximately 20 minutes to complete.  The second time it took about 41 mins to complete and the third time it took approximately 1 hour and 30 mins.
Each time I used exactly the same parameters:
connection.call("ApierV1.LoadTariffPlanFromStorDb", {"TPid":"TARIFFPLAN1", "FlushDb":False, "DryRun":False})

Anyway, I am not sure which method you are using to do inserts into Redis but I saw an interesting post about doing Mass Insertion in Redis which might be useful to you here: http://redis.io/topics/mass-insert

Regards,

Errol
